test/e2e_node/dra_test.go

```diff
@@ -378,6 +420,9 @@ var _ = framework.SIGDescribe("node")("DRA", feature.DynamicResourceAllocation,
 		})
 
 		ginkgo.It("must run pod if NodePrepareResources is in progress for one plugin when Kubelet restarts", func(ctx context.Context) {
+			_, kubeletPlugin2 := start(ctx)
+			kubeletPlugin := newKubeletPlugin(ctx, f.ClientSet, getNodeName(ctx, f), driverName)
+
 			unblock := kubeletPlugin.BlockNodePrepareResources()
```

Comment:
```suggestion
			kubeletPlugin1, kubeletPlugin2 := start(ctx)

			unblock := kubeletPlugin1.BlockNodePrepareResources()
```

I think the old code `kubeletPlugin` is just a typo. the pod below is still refering to the  `kubeletPlugin1Name`. See #126193

BTW, why do you want to replace `BeforeEach` with the start function? I think using `BeforeEach` to initialize resources are pretty common. https://onsi.github.io/ginkgo/#shared-behaviors

Reply:
The problem is that the plugin creates background goroutines within the context of the `BeforeEach`, which gets cancelled when the `BeforeEach` callback returns. That then kills the goroutines before the actual test runs.

The repeated lines are a function call. I think it's okay to repeat that in each `It`.

Reply:
The other reason for removing the `BeforeEach` is that it always started three plugins, even if only two were needed. Now tests can be more specific about what they really need.

This confusion with `kubeletPlugin1` and `kubeletPlugin` is related to that. This test probably didn't work as intended because it blocked the wrong plugin instance, one which wasn't referenced by the pod. I'm testing your suggestion right now and will include it in the next update.

cc @bart0sh

Reply:
> That then kills the goroutines before the actual test runs.

This is a new feature added in this PR.

Reply:
> That then kills the goroutines before the actual test runs

Maybe you can use DeferCleanup, which is specifically documented to run after the test
```golang
ginkgo.BeforeEach(func() {
    ctx, cancel := context.WithCancel(context.Background())
    ginkgo.DeferCleanup(cancel)
    kubeletPlugin := newKubeletPlugin(ctx, f.ClientSet, getNodeName(ctx, f), driverName)
})
```

> The other reason for removing the BeforeEach is that it always started three plugins, even if only two were needed. Now tests can be more specific about what they really need.

I think this is a misunderstanding. `BeforeEach` also only starts needed plugins. The cases under "Resource Kubelet Plugin" only use kubeletPlugin, the cases under "Two resource Kubelet Plugins" only use kubeletPlugin1 and kubeletPlugin2. I've moved the declaration one level deeper to avoid the confusion in #126193.